hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,915 @@
+
+        Returns:
+            dict: A dictionary with the result of the stamping process.
+
+        """
+        # Find the invoice by its ID
+        invoice = request.env["account.move"].browse(invoice_id)
+
+        # Check if the invoice is already stamped
+        if invoice.l10n_mx_edi_cfdi_uuid:
+            return {
+                "message": "The invoice is already stamped.",
+                "UUID": invoice.l10n_mx_edi_cfdi_uuid,
+            }
+
+        # Attempt the stamping process
+        invoice.action_process_edi_web_services()
+
+        # Check again if the invoice was successfully stamped
+        if invoice.l10n_mx_edi_cfdi_uuid:
+            return {
+                "success": "The invoice has been successfully stamped.",
+                "UUID": invoice.l10n_mx_edi_cfdi_uuid,
+            }
+
+        # If stamping failed, get the error messages
+        error_message = invoice.edi_error_message
+        return {
+            "error": "The invoice stamping failed.",
+            "details": error_message,
+        }
+
+    @route(
+        "/send_invoice_by_email/<int:invoice_id>",
+        type="json",
+        auth="user",
+        methods=["POST"],
+    )
+    def send_invoice_by_email(self, invoice_id=False):
+        """
+        Sends an invoice by email.
+
+        This function sends an invoice identified by its ID via email, using the
+        invoice sending wizard in Odoo.
+
+        URL parameter:
+            - invoice_id (int): The ID of the invoice to be sent.
+
+        JSON response:
+            - success (str): A message indicating that the invoice has been successfully sent.
+
+        Returns:
+            dict: A dictionary with a success message.
+
+        """
+        invoice = request.env["account.move"].browse(invoice_id)
+        action = invoice.action_invoice_sent()
+        action_context = action["context"]
+        invoice_send_wizard = (
+            request.env["account.invoice.send"]
+            .with_context(action_context, active_ids=[invoice.id])
+            .create({"is_print": False})
+        )
+
+        # Send the invoice by email
+        invoice_send_wizard.send_and_print_action()
+
+        return {"success": "The invoice has been successfully sent."}
+
+    @route("/confirm_sale_order", methods=["POST"], type="json", auth="user")
+    def confirm_sale_order(self):
+        """
+        Confirms a sale order.
+
+        This function confirms a sale order identified by its ID, using the details
+        provided in the JSON request body.
+
+        JSON request body:
+            - sale_order_id (int): The ID of the sale order to be confirmed.
+
+        JSON response:
+            - message (str): A confirmation message indicating the action performed.
+
+        Returns:
+            dict: A dictionary with a confirmation message.
+
+        """
+        data = request.jsonrequest
+        sale_order_id = data.get("sale_order_id")
```

Comment:
I  would rather avoid this, you can use <model("sale.order"):order>

that way you deacrease code lines, and automatically you got the order id object already as an browseable record set